tox.ini

```diff
@@ -18,8 +18,14 @@ commands =
   yamllint -c .yamllint.yaml tests
 
 [testenv:add_docs]
-deps = git+https://github.com/ansible-network/collection_prep
-commands = collection_prep_add_docs -p .
+setenv =
+  LC_ALL=en_US.UTF-8
+deps =
+  git+https://github.com/ansible-network/collection_prep
+  antsibull-docs
+commands =
+  collection_prep_add_docs -p .
+  antsibull-docs collection-plugins community.vmware --dest-dir docs/ --use-current --output-format simplified-rst --fqcn-plugin-names
```

Comment:
I run this in a container:

```
docker run -it --rm mariolenz/playbook-development
```

And then:

```
root [ /playbooks ]# mkdir -p ~/.ansible/collections/ansible_collections/community
root [ /playbooks ]# git clone https://github.com/mariolenz/community.vmware.git ~/.ansible/collections/ansible_collections/community/vmware
Cloning into '/root/.ansible/collections/ansible_collections/community/vmware'...
remote: Enumerating objects: 12456, done.
remote: Counting objects: 100% (877/877), done.
remote: Compressing objects: 100% (325/325), done.
remote: Total 12456 (delta 598), reused 785 (delta 546), pack-reused 11579
Receiving objects: 100% (12456/12456), 5.90 MiB | 15.34 MiB/s, done.
Resolving deltas: 100% (8300/8300), done.
root [ /playbooks ]# cd ~/.ansible/collections/ansible_collections/community/vmware
root [ ~/.ansible/collections/ansible_collections/community/vmware ]# git checkout antsibull-docs
branch 'antsibull-docs' set up to track 'origin/antsibull-docs'.
Switched to a new branch 'antsibull-docs'
root [ ~/.ansible/collections/ansible_collections/community/vmware ]# LC_ALL="en_US.UTF-8" ansible-playbook tools/update_documentation.yml
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

PLAY [localhost] ***********************************************************************************************************************************

TASK [Gathering Facts] *****************************************************************************************************************************
ok: [localhost]

TASK [Install tox] *********************************************************************************************************************************
changed: [localhost]

TASK [Refresh module documentation and README.md] **************************************************************************************************
changed: [localhost]

PLAY RECAP *****************************************************************************************************************************************
localhost                  : ok=3    changed=2    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   

root [ ~/.ansible/collections/ansible_collections/community/vmware ]# git status
On branch antsibull-docs
Your branch is up to date with 'origin/antsibull-docs'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   README.md
	modified:   docs/community.vmware.vmware_host_inventory_inventory.rst
	modified:   docs/community.vmware.vmware_vm_inventory_inventory.rst

no changes added to commit (use "git add" and/or "git commit -a")
root [ ~/.ansible/collections/ansible_collections/community/vmware ]#
```

This works for me. I don't understand why the CI fails :shrug:

Reply:
In CI it looks like the directory is `/workspace/src/github.com/ansible-collections/community.vmware/`, so it does not end with `/ansible_collections/community/vmware/`. (And likely `ANSIBLE_COLLECTIONS_PATH` isn't set either.)

`collection_prep` links the checkout dir into a temporary directory structure:
```
add_docs run-test: commands[0] | collection_prep_add_docs -p .
INFO      Setting collection name to community.vmware
INFO      Setting GitHub repository url to https://github.com/ansible-collections/community.vmware.git
INFO      /workspace/src/github.com doesn't look enough like a collection
INFO      Temporary collection path /tmp/tmp59mnrwh_ created
INFO      Linking collection to collection path /tmp/tmp59mnrwh_/ansible_collections
INFO      This is required for the Ansible fragment loader to find doc fragments
INFO      Creating namespace directory /tmp/tmp59mnrwh_/ansible_collections/community
INFO      Linking collection /workspace/src/github.com/ansible-collections/community.vmware -> /tmp/tmp59mnrwh_/ansible_collections/community/vmware
INFO      Collection path is /tmp/tmp59mnrwh_/ansible_collections/community/vmware
```
antsibull-docs doesn't do such magic, it assumes - if you provie `--use-current` - that the collection is in a place where ansible can find it.

Reply:
Thanks @felixfontein! I was a little bit confused why it worked for me and not in the CI. But I was testing in `~/.ansible/collections/ansible_collections/community` where ansible can find it. So that's the explanation.

I'll try to think up a solution.

Reply:
Maybe wrap the antsibull-docs call in a bash script that creates a temp directory, copies the checkout into it, puts that path into `ANSIBLE_COLLECTIONS_PATH`, calls antsibull-docs, and cleans up.

I think it would be good to have a tool which does this, but I'm right now not sure where it should live. Something like that is also useful for running ansible-test or other tools that expect that the checkout lives in a specific directory structure and/or is available through `ANSIBLE_COLLECTIONS_PATH`.

Reply:
> I think it would be good to have a tool which does this, but I'm right now not sure where it should live. Something like that is also useful for running ansible-test or other tools that expect that the checkout lives in a specific directory structure and/or is available through `ANSIBLE_COLLECTIONS_PATH`.

Actually, I was thinking something similar today. I thought it might be useful to have an ansibleenv similar to a pyenv. I could install the checked out collection in it and remove it after whatever I've done (like running antsibull-docs). I don't know if this is really possible, or even feasible.

For now, I'll try to wrap antsibull-docs in a script that more or less mimics what collection_prep does. Thanks for the suggestion @felixfontein!